Content.Server/_Goobstation/Heretic/Ritual/CustomBehavior.Sacrifice.cs

```diff
@@ -118,10 +187,123 @@ public override void Finalize(RitualData args)
                 && isCommand)
                     crewHeadObjComp.Sacrificed += 1;
             }
+
         }
 
         // reset it because it refuses to work otherwise.
         uids = new();
         args.EntityManager.EventBus.RaiseLocalEvent(args.Performer, new EventHereticUpdateTargets());
     }
+
+    //# funkystation
+    //# sacrifice function to safely teleport them away -space
+    private void SafeSacrifice(TransformComponent transform, RitualData args, EntityUid uid)
+    {
+
+        //drop all items -space
+
+
+        if (args.EntityManager.TryGetComponent<InventoryComponent>(uid, out var comp2))
+        {
+            var transformSystem = _system.EntityManager.System<TransformSystem>();
+            var inventorySystem = _system.EntityManager.System<InventorySystem>();
+            var sharedPopupSystem = _system.EntityManager.System<SharedPopupSystem>();
+
+            foreach (var item in inventorySystem.GetHandOrInventoryEntities(uid))
+            {
+                transformSystem.DropNextTo(item, uid);
+            }
+        }
+
+
+
+
+        // teleports them away -space
+        var mapPos = _xform.GetWorldPosition(transform);
+        var radius = 30;
+        var gridBounds = new Box2(mapPos - new Vector2(radius, radius), mapPos + new Vector2(radius, radius));
+        if (args.EntityManager.TryGetComponent<MobStateComponent>(uid, out var mobstate))
+
+        {
+            if (mobstate.CurrentState == Shared.Mobs.MobState.Dead)
+            {
+
+                //check if DEAD
+
+                var ent = uid;
+                var check = 1;
+
+                do //loop here to place the person in the right spot that is on a grid (so they cant get lost in space) -space
```

Comment:
Just verify with me that you've seen the way teleports are handled in vanilla, since I don't want us to be reinventing the wheel here.

Reply:
Also, putting this in a separate function would be great.